extjsdk/src/main/java/io/vantiq/extjsdk/Utils.java

```diff
@@ -1,114 +1,127 @@
 package io.vantiq.extjsdk;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.util.Properties;
 
 public class Utils {
 
+    // String used by methods to synch on
+    private static final String SYNCH_LOCK = "sycnhLockString";
+
     public static final String SEND_PING_PROPERTY_NAME = "sendPings";
     public static final String PORT_PROPERTY_NAME = "tcpProbePort";
     public static final String SERVER_CONFIG_DIR = "serverConfig";
     public static final String SERVER_CONFIG_FILENAME = "server.config";
     public static final String SECRET_CREDENTIALS = "CONNECTOR_AUTH_TOKEN";
 
-    private static final Logger log = LoggerFactory.getLogger(Utils.class);
+    // The properties object containing the data from the server configuration file
+    public static Properties serverConfigProperties;
 
     public static Properties obtainServerConfig() {
         return obtainServerConfig(SERVER_CONFIG_FILENAME);
     }
-    
+
     /**
      * Turn the given configuration file into a {@link Properties} object.
      *
      * @param fileName  The name of the configuration file holding the server configuration.
      * @return          The properties specified in the file.
      */
     public static Properties obtainServerConfig(String fileName) {
-        File configFile = new File(SERVER_CONFIG_DIR, fileName);
-        Properties properties = new Properties();
+        synchronized (SYNCH_LOCK) {
+            File configFile = new File(SERVER_CONFIG_DIR, fileName);
+            serverConfigProperties = new Properties();
 
-        try {
-            if (!configFile.exists()) {
-                configFile = new File(fileName);
-            }
-            properties.load(Files.newInputStream(configFile.toPath()));
-
-            // Next we check for the existence of an environment variable containing a secret reference to the authToken
-            // We only set it if the value is not empty and if the authToken wasn't already specified in the
-            // server.config
-            String secretAuthToken = System.getenv(SECRET_CREDENTIALS);
-            if (secretAuthToken != null && !secretAuthToken.trim().isEmpty()
-                    && properties.getProperty("authToken") == null) {
-                properties.setProperty("authToken", secretAuthToken);
+            try {
+                if (!configFile.exists()) {
```

Comment:
Why are we "creating" an empty file in this case?  Seems like we just leave the properties empty & return them or null -- whichever is expected.  Or throw an error. This was probably leftover code I copied, but not sure what it's attempting to do here...

Reply:
I think the idea here is that we first check to see if the file exists in the `SERVER_CONFIG_DIR`, and then if not we assume it's just in the workingdir. If the file isn't present at all, then we'll throw a RuntimeException (by catching the IOException).